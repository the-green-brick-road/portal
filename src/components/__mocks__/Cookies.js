/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
