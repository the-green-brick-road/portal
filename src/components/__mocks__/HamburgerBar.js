/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
