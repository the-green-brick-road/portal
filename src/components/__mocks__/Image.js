/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
