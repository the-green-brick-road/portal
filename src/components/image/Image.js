/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
