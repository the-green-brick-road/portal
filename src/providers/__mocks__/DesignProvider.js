/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Design provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

/* Material UI includes */
import { CssBaseline }                                 from '@mui/material';
import { ThemeProvider, createTheme }                  from '@mui/material/styles';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isWebpSupported: true,
    isSliding:       false,
    isDarkMode:      false,
    images:          {},
    screen:          '',
    sizes:           {},
    setIsSliding:    jest.fn((value) => { state.isSliding = value }),
};

function DesignProvider(props) {

    const { children, isWebpSupported, screen, isDarkMode = false, isSliding = false, sizes, images = [], theme = {} } = props;

    /* ------ Defining behaviour ------- */
    var local_theme = {}
    if (Object.keys(theme).length !== 0) { local_theme = createTheme(theme); }

    state.isWebpSupported = isWebpSupported;
    state.screen          = screen;
    state.images          = {}
    state.sizes           = sizes;
    state.isSliding       = isSliding;
    state.isDarkMode      = isDarkMode;
    for (const image of images)
    {

        const rawext = image["raw"].split('.').pop();
        const webext = image["web"].split('.').pop();

        state.images[image.name] = {}
        state.images[image.name][rawext] = {}
        state.images[image.name][webext] = {}

        state.images[image.name][rawext]['small']           = {}
        state.images[image.name][rawext]['small']['img']    = require(`../../assets/small/${image.raw}`)
        state.images[image.name][rawext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][rawext]['medium']          = {}
        state.images[image.name][rawext]['medium']['img']   = require(`../../assets/medium/${image.raw}`)
        state.images[image.name][rawext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][rawext]['large']           = {}
        state.images[image.name][rawext]['large']['img']    = require(`../../assets/${image.raw}`)
        state.images[image.name][rawext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name][webext]['small']           = {}
        state.images[image.name][webext]['small']['img']    = require(`../../assets/small/${image.web}`)
        state.images[image.name][webext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][webext]['medium']          = {}
        state.images[image.name][webext]['medium']['img']   = require(`../../assets/medium/${image.web}`)
        state.images[image.name][webext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][webext]['large']           = {}
        state.images[image.name][webext]['large']['img']    = require(`../../assets/${image.web}`)
        state.images[image.name][webext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name]['default']= require(`../../assets/${image.raw}`)

    }

    /* ----------- Define HTML --------- */
    return (
        <ThemeProvider theme={local_theme}>
            <CssBaseline />
            <context.Provider value={state}>
                {children}
            </context.Provider>
        </ThemeProvider>
    );

}

function useDesign() { return state; };

export { useDesign, DesignProvider };