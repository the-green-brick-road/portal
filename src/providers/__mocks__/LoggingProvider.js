/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Logging provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isLoggingActivated : true,
    logText: jest.fn((component, level, topic, message) => {  console.log(`${component} - ${level} - ${topic} -${message}`) }),
};

function LoggingProvider(props) {

    const { children } = props;

    /* ----------- Define HTML --------- */
    return (
        <context.Provider value={state}>
            {children}
        </context.Provider>
    );

}

function useLogging() { return state; };

export { useLogging, LoggingProvider };