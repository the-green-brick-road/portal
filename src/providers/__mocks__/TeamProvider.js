/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Team provider mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @14 june 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = { team: {} };

function TeamProvider(props) {

    const { children, team = {} } = props;

    /* ------ Defining behaviour ------- */
    state.team = team;

    /* ----------- Define HTML --------- */
    return (
        <context.Provider value={state}>
            {children}
        </context.Provider>
    );

}

function useTeam() { return state; };

export { useTeam, TeamProvider };