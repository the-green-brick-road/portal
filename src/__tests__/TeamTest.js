/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Home view component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @30 may 2023
# Latest revision: 30 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM }       from '@testing-library/react'
import { expect, test}             from '@jest/globals';

/* Component under test */
import { default as Team }       from '../views/team/Team';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { default as MockImage }    from '../components/__mocks__/Image';
import { useTeam as mockUseTeam, TeamProvider as MockTeamProvider } from '../providers/__mocks__/TeamProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../providers/__mocks__/LoggingProvider';
import { useDesign as mockUseDesign, DesignProvider as MockDesignProvider } from '../providers/__mocks__/DesignProvider';
jest.mock("../components", () => ({ Image: (props) => MockImage(props) }));
jest.mock('../providers', () => ({
    useTeam: (() => { return mockUseTeam(); }),
    useLogging: (() => { return mockUseLogging(); }),
    useDesign: (() => { return mockUseDesign(); }),
}));
/* eslint-enable jest/no-mocks-import */

describe("Team view" ,() => {

    test('Should display team page', async () => {

        const Members = {
            1 : {name : 'member1', id : '1', grade : 8, start:{seconds:4000000}, coding : 2, building : 2, description : 'member1'},
            2 : {name : 'member2', id : '2', grade : 7, start:{seconds:6000000}, coding : 1, building : 1, description : 'member2'},
            3 : {name : 'member3', id : '3', grade : -1, start:{seconds:3000000}, end : {seconds:2000000}, coding : 1, building : 2, description : 'member3'},
        }

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockDesignProvider>
                        <MockTeamProvider team={Members}>
                            <Team/>
                        </MockTeamProvider>
                    </MockDesignProvider>
                </MockLoggingProvider>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

})
