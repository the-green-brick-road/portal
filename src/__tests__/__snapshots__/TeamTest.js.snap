// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team view Should display team page 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"width: 100%; padding: 0px; position: relative;\\"
      >
        <div
          class=\\"MuiBox-root css-0\\"
          style=\\"width: 100%; pointer-events: none; background-color: blue;\\"
        />
      </div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"width: 100%; overflow-x: hidden; overflow-y: hidden;\\"
      >
        <h1
          class=\\"MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root\\"
        >
          Our team
        </h1>
        <div
          class=\\"MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root\\"
        >
          <div
            class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-24 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 css-14j1sok-MuiGrid-root\\"
            style=\\"padding: 0px;\\"
          >
            <div
              class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root\\"
              style=\\"background-color: rgb(238, 238, 238); margin: 10px;\\"
            >
              <div
                class=\\"MuiCardHeader-root css-185gdzj-MuiCardHeader-root\\"
                style=\\"background-color: rgb(238, 238, 238); padding: 5px 5px 5px 5px; text-align: center;\\"
              >
                <div
                  class=\\"MuiCardHeader-content css-1qbkelo-MuiCardHeader-content\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-6qyd8u-MuiTypography-root\\"
                  >
                    member1
                  </span>
                </div>
              </div>
              <div
                class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"
                style=\\"background-color: rgb(238, 238, 238); padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"color: rgb(0, 0, 0); font-size: 14px; font-style: italic; padding-bottom: 10px;\\"
                >
                  \\"
                  member1
                  \\"
                </p>
                <div
                  class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root\\"
                >
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Member : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(61, 237, 151); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        1970
                         - 
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Grade : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(112, 125, 59); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        8
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Coding : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(132, 205, 84); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        2
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Building : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(133, 151, 128); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        2
                         
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-24 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 css-14j1sok-MuiGrid-root\\"
            style=\\"padding: 0px;\\"
          >
            <div
              class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root\\"
              style=\\"background-color: rgb(238, 238, 238); margin: 10px;\\"
            >
              <div
                class=\\"MuiCardHeader-root css-185gdzj-MuiCardHeader-root\\"
                style=\\"background-color: rgb(238, 238, 238); padding: 5px 5px 5px 5px; text-align: center;\\"
              >
                <div
                  class=\\"MuiCardHeader-content css-1qbkelo-MuiCardHeader-content\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-6qyd8u-MuiTypography-root\\"
                  >
                    member2
                  </span>
                </div>
              </div>
              <div
                class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"
                style=\\"background-color: rgb(238, 238, 238); padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"color: rgb(0, 0, 0); font-size: 14px; font-style: italic; padding-bottom: 10px;\\"
                >
                  \\"
                  member2
                  \\"
                </p>
                <div
                  class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root\\"
                >
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Member : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(61, 237, 151); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        1970
                         - 
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Grade : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(112, 125, 59); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        7
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Coding : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(132, 205, 84); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        1
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Building : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(133, 151, 128); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        1
                         
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"width: 100%; overflow-x: hidden; overflow-y: hidden;\\"
      >
        <h1
          class=\\"MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root\\"
        >
          Our alumni
        </h1>
        <div
          class=\\"MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root\\"
        >
          <div
            class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-24 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 css-14j1sok-MuiGrid-root\\"
            style=\\"padding: 0px;\\"
          >
            <div
              class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root\\"
              style=\\"background-color: rgb(238, 238, 238); margin: 10px;\\"
            >
              <div
                class=\\"MuiCardHeader-root css-185gdzj-MuiCardHeader-root\\"
                style=\\"background-color: rgb(238, 238, 238); padding: 5px 5px 5px 5px; text-align: center;\\"
              >
                <div
                  class=\\"MuiCardHeader-content css-1qbkelo-MuiCardHeader-content\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-6qyd8u-MuiTypography-root\\"
                  >
                    member3
                  </span>
                </div>
              </div>
              <div
                class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"
                style=\\"background-color: rgb(238, 238, 238); padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"color: rgb(0, 0, 0); font-size: 14px; font-style: italic;\\"
                >
                  \\"
                  member3
                  \\"
                </p>
                <div
                  class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root\\"
                >
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Member : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(61, 237, 151); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        1970
                         - 
                        1970
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Coding : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(132, 205, 84); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        1
                         
                      </p>
                    </div>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <p
                      class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                      style=\\"color: rgb(0, 0, 0); font-size: 14px;\\"
                    >
                       Building : 
                    </p>
                  </div>
                  <div
                    class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root\\"
                  >
                    <div
                      class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root\\"
                      style=\\"background-color: rgb(133, 151, 128); padding-left: 5px; padding-right: 5px;\\"
                    >
                      <p
                        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                        style=\\"color: rgb(255, 255, 255); font-size: 14px; font-weight: bold; text-align: center;\\"
                      >
                         
                        2
                         
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>"
`;
